### PR TITLE
Extensions homogenization, cleaned up.

### PR DESCRIPTION
**Pull request by Drup**
Rebased and cleaned up version of extension homogenisation. I will give some comments that should be fixed before merge now.

### CODE REVIEW

File: src/server/ocsigen_extensions.ml [hunk 1]
```diff
@@ -57,7 +57,26 @@ let badconfig fmt = Printf.ksprintf (fun s -> raise (Error_in_config_file s)) fm
 
 (*****************************************************************************)
 (* virtual hosts: *)
-type virtual_hosts = (string * Netstring_pcre.regexp * int option) list
+
+module VirtualHost = struct
```
Comment (rgrinberg) [inline src/server/ocsigen_extensions.ml]: Isn't `Virtual_host` more consistent?

File: src/server/ocsigen_extensions.ml [hunk 2]
```diff
@@ -57,7 +57,26 @@ let badconfig fmt = Printf.ksprintf (fun s -> raise (Error_in_config_file s)) fm
...
+    ?port
+    () = (host, pattern, port)
+
+  let hash (host, _, port) = Hashtbl.hash (host, port)
+  let equal (h1, _r1, p1) (h2, _r2, p2) =
+    h1 = h2 && p1 = p2
+
+  let host (host, _, _) = host
+  let pattern (_, pattern, _) = pattern
+  let port (_, _, port) = port
```
Comment (rgrinberg) [inline src/server/ocsigen_extensions.ml]: Might be worth to consider just making it a record. Don't how much breakage this would cause.

Comment (dinosaure) [inline src/server/ocsigen_extensions.ml]: Yeah, I think it´s better to use record.

File: src/server/ocsigen_extensions.ml (hunk 1 above)
Comment (dinosaure) [inline src/server/ocsigen_extensions.ml]: If I remember correctly, I abstract ´Virtual_host´ to a problem of dependence (between configuration file and configuration in OCaml).

**Comment by vasilisp** [comment]
Rebase / reopen if still relevant.

**Comment by Drup** [comment]
While I didn't kept working on it, that one was useful and improved the codebase significantly .. :/

**Comment by vasilisp** [comment]
@Drup, so where some of the others, but I doubt anybody is available to get them in mergeable shape now. I guess we can revisit some of these concerns in the process of restructuring around Cohttp.

**Pull request closed without merging**